pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -110,16 +110,20 @@ type TidbClusterList struct {
 // TidbClusterSpec describes the attributes that a user creates on a tidb cluster
 type TidbClusterSpec struct {
 	// Discovery spec
-	Discovery DiscoverySpec `json:"discovery,omitempty"`
+	// +optional
+	Discovery *DiscoverySpec `json:"discovery,omitempty"`
```

Comment:
No need to update this as there is only resource configurations in this field.

Reply:
ok